### DISCUSSION TITLE
Why is there no control sequence for positioning an image at specific cell coordinates?

### DISCUSSION THREAD

**Discussion by slavistan** [kovidgoyal/kitty#6151] [Q&A] [answered]
Hi!

I've spent the weekend playing around with kitty and, liking what I see so far, I am now in the midst of integrating my `lf` setup with kitty. I have attempted to use `icat` to achieve this, but as it stands it can't control an image's placement id when drawing an image and `icat --clear` always deletes _all_ placements indiscriminately, whereas in need to be selective in certain cases. I'll create a ticket or pull request for that once I've gathered a little more data and understanding.

I have thus attempted to work at the protocol level itself, `printf`ing my way close to victory. Transferring images to the terminal turns out to be trivial, especially so for the local case, and even displaying images at the cursor position is straightforward. However, I've come to find out that there's no instruction at the protocol level that would allow me to specify the position in cell coordinates of _where_ to display an image or animation. While I can specify the width and height in cell coordinates, there's no corresponding instruction for the position and I wonder why that is.

May I kindly ask someone in the know to explain this to me? Thank you.

**Comment by kovidgoyal** [comment, marked as answer]
The image is displayed at the current cursor position. There already
exist escape codes to move the cursor around, as well as to save and
restore the cursor position. Use those.

**Reply by iambumblehead** [reply]
When using bash scripts with ncurses applications, escape codes are unusable. The reason is ncurses takes control of /dev/stdout and /dev/stderr used to message the escape codes. It is possible to delay updates at the ncurses ui to allow a bash script to send escape codes through /dev/tty instead, but this is inconvenient and requires coordination with the parent ncurses application.

It would be great if kitty icat could display images to specific sizes without explicit cursor position.

Yes, images can be pre-processed to the specific size of the view area offered by the ncurses app, however, it would be a great help if it were not necessary.

**Reply by kovidgoyal** [reply]
kitty icat is not magic it will display images by using escape codes as well. So it would still need co-ordiantion with ncurses. If your host application doesnt provide a safe way to do tty IO that would be a feature request for your host application.
